esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6590,5 +6591,104 @@ bool CWsDfuEx::onDFUFilePublish(IEspContext &context, IEspDFUFilePublishRequest
+            {
+                RemoteFilename rfn;
+                part.getFilename(c, rfn);
+                try
+                {
+                    Owned<IFile> file = createIFile(rfn);
+                    if (compressed)
+                    {
+                        compressedSize = file->size();
+                        Owned<IFileIO> io = createCompressedFileReader(file);
+                        if (io)
+                            size = io->size();
+                    }
+                    else
+                    {
+                        size = file->size();
+                    }
+                }
+                catch (IException *e)
+                {
+                    VStringBuffer tmp("%s: ", lfn.get());
+                    rfn.getPath(tmp);
+                    EXCLOG(e, tmp.str());
+                    e->Release();
+
+                    size = (offset_t)-1;
+                    compressedSize = (offset_t)-1;
+                }
+                if ((size != (offset_t)-1) && (!compressed || (compressedSize != (offset_t)-1)))
+                    break;
+            }
+            return 0;
+        };
+
+        offset_t getSize() { return size; }
+        offset_t getCompressedSize() { return compressedSize; }
+    };
+
+    //Get the sizes and the compressed sizes for each file parts in parallel
+    CIArrayOf<CFilePartSizeThread> threads;
+    bool compressed = fileDesc->isCompressed();
+    Owned<IPartDescriptorIterator> pi = fileDesc->getIterator();
+    ForEach(*pi)
+    {
+        CFilePartSizeThread* thread = new CFilePartSizeThread(lfn, pi->query(), compressed);
+        threads.append(*thread);
+        thread->start();
+    }
+    ForEachItemIn(i, threads)
+        threads.item(i).join();
+
+    //Set the size and the compressed size for each file parts and calculate file compressed size.
+    offset_t fileCompressedSize = (offset_t)0;
+    ForEachItemIn(ii, threads)
+    {
+        CFilePartSizeThread& thread = threads.item(ii);
+        offset_t size = thread.getSize();
+        offset_t compressedSize = thread.getCompressedSize();
+        if ((size == (offset_t)-1) || (compressed && (compressedSize == (offset_t)-1)))
+        {
+            if (compressed)
+                fileCompressedSize = (offset_t)-1;
+            IERRLOG("Failed to get part size for file %s part %u", lfn, ii);
+            break;
+        }
+
+        IPartDescriptor *partDesc = fileDesc->queryPart(ii);
+        IPropertyTree &props = partDesc->queryProperties();
+        props.setPropInt64("@size", size);
+        if (compressed)
+        {
+            props.setPropInt64("@compressedSize", compressedSize);
+            fileCompressedSize += compressedSize;
+        }
+    }
+
+    //Set file compressed size
+    if (compressed && (fileCompressedSize != (offset_t)-1))
+    {
+        fileDesc->queryProperties().setPropInt64("@compressedSize", fileCompressedSize);
```

Comment:
I think (as with top-level @size), @compressedSize should be auto-computed/populated, as long as parts have this property set. If I'm right, this code is not necessary.
If I'm wrong, I can't see the code that sets it in Thor for example, so may imply a bug.
Can you check?

Reply:
Looks like the auto-computed/populated does not work here. If I do not set the compressedSize here, the Attr/@compressedSize is not set for the file. But, when I create a compressed file using ECL's OUTPUT option ENCRYPT, the Attr/@compressedSize is set.

Reply:
Inside the CDistributedFile constructor, there is some code to calculate/set the @size. But, there is no code for the @compressedSize. Is that a bug?

Reply:
Yes it is really, or an opportunity to improve and be more consistent. The engines hit the issue, by manually setting compressedSize (and sometimes size) but not consistently.
I've released a fix : https://track.hpccsystems.com/browse/HPCC-25708